app/models/user.rb

```diff
@@ -532,4 +538,33 @@ def update_pseud_name
    def log_change_if_login_was_edited
      create_log_item( options = {:action => ArchiveConfig.ACTION_RENAME, :note => "Old Username: #{login_was}; New Username: #{login}"}) if login_changed?
    end
+
+  def perform(operation)
+    case operation
+    when 'update_last_login'
+      self.update_last_login
+    end
+  end
+
+  # called from rake
+  def self.update_last_login
```

Comment:
You don't want this to be private since it's called from rake so would suggest moving this method somewhere else in the file out from under the private declaration.

Ideally tbh this should go into a separate module and not be in the already horribly bloated user.rb class.